redfish-core/lib/assembly.hpp

```diff
@@ -532,6 +538,215 @@ inline void getChassis(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
         std::array<const char*, 2>{"xyz.openbmc_project.Inventory.Item.Chassis",
                                    "xyz.openbmc_project.Inventory.Item.Board"});
 }
+
+/**
+ * @brief API used to fill the Assembly id of the assembled object that
+ *        assembled in the given assembly parent object path.
+ *
+ *        bmcweb using the sequential numeric value by sorting the
+ *        assembled objects instead of the assembled object dbus id
+ *        for the Redfish Assembly implementation.
+ *
+ * @param[in] aResp - The redfish response to return.
+ * @param[in] assemblyParentServ - The assembly parent dbus service name.
+ * @param[in] assemblyParentObjPath - The assembly parent dbus object path.
+ * @param[in] assemblyParentIface - The assembly parent dbus interface name
+ *                                  to valid the supports in the bmcweb.
+ * @param[in] assemblyUriPropPath - The redfish property path to fill with id.
+ * @param[in] assembledObjPath - The assembled object that need to fill with
+ *                               its id. Used to check in the parent assembly
+ *                               associations.
+ * @param[in] assembledUriVal - The assembled object redfish uri value that
+ *                              need to replace with its id.
+ *
+ * @return The redfish response with assembled object id in the given
+ *         redfish property path if success else returns the error.
+ */
+inline void fillWithAssemblyId(
+    const std::shared_ptr<bmcweb::AsyncResp>& aResp,
+    const std::string& assemblyParentServ,
+    const sdbusplus::message::object_path& assemblyParentObjPath,
```

Comment:
include what you use ?

Reply:
Thank you for your valid comments. We have created the JIRA ticket https://jsw.ibm.com/browse/PFEBMC-620 to address this comment.
The scope of this task was to cherry pick the working HWIsolation commits from 1030. And as the number of commits are more, it is difficult to refactor them to align with upstream as part of this PR. So we will address all them as a separate story.